server/paymentProviders/opencollective/host.js

```diff
@@ -20,19 +20,29 @@ paymentMethodProvider.refundTransaction = async (transaction, user) => {
     throw new Error('Cannot refund a transaction from the same collective');
   }
 
-  const host = await models.Collective.findByPk(transaction.CollectiveId);
+  if (transaction.type === TransactionTypes.DEBIT) {
+    transaction = await transaction.getRelatedTransaction({ type: TransactionTypes.CREDIT });
+  }
+
+  if (!transaction) {
+    throw new Error('Cannot find any CREDIT transaction to refund');
+  } else if (transaction.RefundTransactionId) {
+    throw new Error('This transaction has already been refunded');
+  }
+
+  const collective = await models.Collective.findByPk(transaction.CollectiveId);
 
-  const balance = await host.getBalanceWithBlockedFunds();
+  const balance = await collective.getBalanceWithBlockedFunds({ currency: transaction.currency });
   if (balance < transaction.amount) {
     throw new Error(
       `Not enough funds available (${formatCurrency(
         balance,
-        host.currency,
+        collective.currency,
```

Comment:
If the balance is expressed  in `transaction.currency` then why format it with `collective.currency` ?

Reply:
It's a mistake; corrected it. Thanks. 👍🏽